libs/platform/src/lib/components/form/step-input/base.step-input.ts

```diff
@@ -0,0 +1,477 @@
+import { ChangeDetectorRef, Input, Output, EventEmitter, Renderer2, Directive, OnInit } from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+import { BehaviorSubject } from 'rxjs';
+import { takeUntil, switchMap, map } from 'rxjs/operators';
+
+import { RtlService } from '@fundamental-ngx/core';
+
+import { BaseInput } from '../base.input';
+import { ContentDensity } from '../form-control';
+import { StepInputConfig } from './step-input.config';
+import { addAndCutFloatingNumberDistortion, getNumberDecimalLength } from './step-input.util';
+
+/** Change event object emitted by Platform Step Input component */
+export abstract class StepInputChangeEvent<T extends StepInputComponent = StepInputComponent, K = number> {
+    constructor(
+        /** The source Step Input of the event. */
+        public source: T,
+        /** The new value of a control. */
+        public payload: K
+    ) {}
+}
+
+export const enum StepInputAlign {
+    Left = 'left',
+    Center = 'center',
+    Right = 'right'
+}
+
+export type StepInputStepFunctionAction = 'increase' | 'decrease';
+export type StepInputStepFunction = (value: number, action: StepInputStepFunctionAction) => number;
+
+const ALIGN_INPUT_OPTIONS_LIST = [StepInputAlign.Left, StepInputAlign.Center, StepInputAlign.Right];
+
+/**
+ * StepInputComponent is a base abstract class that should be used
+ * to create type specific StepInput components such as number, money, unitOfMeasure
+ * This holds base Step Input functionality that can be abstracted
+ */
+@Directive()
+export abstract class StepInputComponent extends BaseInput implements OnInit {
+    /** content Density of element. cozy | compact */
+    @Input()
+    set contentDensity(contentDensity: ContentDensity) {
+        this._contentDensity = contentDensity;
+        this.isCompact = this._contentDensity === 'compact';
+    }
+
+    /** Sets input value */
+    @Input()
+    get value(): number {
+        return super.getValue();
+    }
+    set value(value: number) {
+        if (value !== this._value) {
+            super.setValue(value);
+            this._calculateCanDecrementIncrement();
+        }
+    }
+
+    /** Sets minimum value boundary */
+    @Input()
+    set min(min: number) {
+        this._min = !isNaN(min) ? min : -Number.MAX_VALUE;
+        this._calculateCanDecrement();
+    }
+    get min(): number {
+        return this._min;
+    }
+
+    /** Sets maximum value boundary */
+    @Input()
+    set max(max: number) {
+        this._max = !isNaN(max) ? max : Number.MAX_VALUE;
+        this._calculateCanIncrement();
```

Comment:
In base input there is `stateChanges` and you need to emit event up to parent  (form-field) which listens for changes and detect chances once.

Reply:
`max/min` options do not change control value nor change error state, they enable/disable ability to increment/decrement.
If so it's not clear for me why we should trigger event in the `stateChanges` subject.